File: Top Interview Questions/Easy Collection/Array/Two Sum.py
"""
URL : https://leetcode.com/explore/interview/card/top-interview-questions-easy/92/array/546/
"""
class Solution:
    def twoSum(self, nums, target: int):
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if (nums[i] + nums[j] == target):
                    return [i,j]

    def twoSumHash(self, nums, target):
        dic = {}
        for index, num in enumerate(nums):
            complement = target - num
            if dic.get(complement) is not None and dic.get(complement) != index:
                return [index, dic.get(complement)]
            dic[num] = index

Solution().twoSumHash([2,7,11,15], 9)

File: Top Interview Questions/Easy Collection/Array/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = {0,0};

        for (int i = 0; i < nums.length; i++) {
            for (int j = i+1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    result = new int[] {i,j};
                }
            }
        }

        return result;
    }

    public int[] twoSumHash(int[] nums, int target) {
        int[] result = new int[2];
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (map.get(complement) != null && map.get(complement) != i) {
                result[0] = i;
                result[1] = map.get(complement);
                break;
            }
            map.put(nums[i], i);
        }

        return result;
    }
}